onnxruntime/core/providers/webgpu/math/cum_sum.cc

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/math/cum_sum.h"
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    CumSum,
+    kOnnxDomain,
+    11, 13,
+    kWebGpuExecutionProvider,
+    (*KernelDefBuilder::Create()).TypeConstraint("T", WebGpuSupportedFloatTypes()).TypeConstraint("T2", WebGpuSupportedNumberTypes()),
```

Comment:
Need to map the second input to CPU in order to retrieve axis in the C++ code. Please check
https://github.com/microsoft/onnxruntime/blob/ae501eeb4d72e2f81d92ad8f7ede4d666bd3902b/onnxruntime/core/providers/js/operators/cumsum.cc#L19